app/frontend/javascript/utils/google-analytics/index.test.js

```diff
@@ -50,4 +55,90 @@ describe('google_tag.mjs', () => {
       expect(document.cookie).toContain('analytics_consent=true')
     })
   })
+
+  describe('sendPageViewEvent()', () => {
+    describe('when the dataLayer array is not already present on the window object', () => {
+      beforeEach(() => {
+        window.dataLayer = undefined
+      })
+
+      it('creates the dataLayer array and pushes a pageView event', function () {
+        sendPageViewEvent()
+        expect(window.dataLayer).toContainEqual({
+          event: 'page_view',
+          page_view: {
+            location: window.location,
+            referrer: '',
+            schema_name: 'simple_schema',
+            status_code: 200,
+            title: ''
+          }
+        })
+      })
+    })
+    describe('when the dataLayer array is already present on the window object', () => {
+      const existingDataLayerObject = {
+        data: 'Some existing data in the dataLayer'
+      }
+
+      beforeEach(() => {
+        window.dataLayer = [existingDataLayerObject]
+      })
+
+      it('the existing dataLayer content is preserved', function () {
+        sendPageViewEvent()
+        expect(window.dataLayer).toContainEqual(existingDataLayerObject)
+      })
+
+      it('the pageView event is pushed to the dataLayer', function () {
+        sendPageViewEvent()
+        expect(window.dataLayer).toContainEqual({
+          event: 'page_view',
+          page_view: {
+            location: window.location,
+            referrer: '',
+            schema_name: 'simple_schema',
+            status_code: 200,
+            title: ''
+          }
+        })
+      })
+    })
+  })
+
+  describe('attachExternalLinkTracker()', () => {
+    const targetLinkText = 'A link to example.com'
+    const targetLinkUrl = 'http://example.com/'
+
+    const existingDataLayerObject = {
+      data: 'Some existing data in the dataLayer'
+    }
+
+    beforeEach(() => {
+      window.document.body.innerHTML = `<a href="${targetLinkUrl}">${targetLinkText}</a>`
+      window.dataLayer = [existingDataLayerObject]
+    })
+
+    it('the existing dataLayer content is preserved', function () {
+      attachExternalLinkTracker()
+      document.querySelector('a').click()
+      expect(window.dataLayer).toContainEqual(existingDataLayerObject)
+    })
+
+    it('the pageView event is pushed to the dataLayer', function () {
```

Comment:
Should this be "the navigation event"?

```suggestion
    it('the navigation event is pushed to the dataLayer', function () {
```

Reply:
Thanks @stephencdaly , good spot! have updated